### ISSUE TITLE
Raise if script/setup has never been run

### ISSUE THREAD

**Issue by paulcsmith** [luckyframework/lucky_cli#504]
Some people skip the script/setup and then their system is not working. We should add something to script/system_check that  before anything else checks that `script/setup` has been run. This will also encourage people to use `script/setup` and keep it up to date!

The way I think we can do this is by touching a file at the end of script/setup `touch .lucky_has_been_setup` or something like that. Then we add that to the `.gitignore` and add `script/system_check` that looks for that file and prints that they should run `script/setup`

Thoughts @jwoertink @edwardloveall ?

**Comment by edwardloveall** [comment]
I dig it, but the ignored file I think will have its own problems because its not actually the thing lucky needs to boot. Is there a lighter-weight check we could run every time? Look for node, crystal, and lucky and a db connection or something?

Or what about if lucky dev fails on startup at all, it recommend `script/setup`? (do we do that already?) I'm mostly just worried that the file will get out of sync with whether or not the user is _actually_ setup.

**Comment by paulcsmith** [comment]
>  I'm mostly just worried that the file will get out of sync with whether or not the user is actually setup.

That is a really good point. I think we need to figure something out for this because that could get annoying.

We can't really check for certain things because that is what `system_check` does, but that may not catch everything. There's stuff in script/setup like migrating the db, install yarn deps, etc. Also people may add more stuff to it (setting up secrets, etc.) So it'd be ideal IMO to somehow make sure the actual setup script was run.

Maybe we can figure something out that meets both criteria... 🤔

**Comment by paulcsmith** [comment]
Idea!!

What if we use the ignored file, but give the tip in the error message to run `touch .{whatever_file_we_decide}` to ignore the message? Seems like a simple way to get around it. And I think for the most part that file will stay there unless they delete the repo and re-clone or something. In which case they probably need to run setup anyway

Thoughts?

**Comment by edwardloveall** [comment]
Yeah I think that would work, or it's at least worth trying 🙂 I keep thinking about just trying to check things instead of setup things, but it gets complex really fast. I think this is the better trade off.
